### PR TITLE
Show product listing

### PR DESCRIPTION
**Pull request by rupertlion**
## PT Link

https://www.pivotaltracker.com/story/show/160720777

## Included in PR:

* Passing feature tests for product listing including name, description and price (as float)
* Added route and controller for Products
* Products model created and migrated
* Passing all feature and unit tests for validation and show of product listings

## What we've learnt

* Take it slow - don't rush

### CODE REVIEW

**Review by SophieSjoberg — APPROVED** [review]
Well done! However - need to see more git ponging action here.

File: app/models/product.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class Product < ApplicationRecord
+    validates :name, presence: true
+    validates :description, presence: true
+    validates :price, presence: true
```
Comment (tochman) [inline app/models/product.rb:4]:
```
validates_presence_of :name, :description, :price
```

File: config/routes.rb [hunk 2]
```diff
@@ -1,3 +1,4 @@
 Rails.application.routes.draw do
-  # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
+  get 'products/index'
```
Comment (tochman) [inline config/routes.rb:2]: Remove the auto-generated routes or send `--skip-routes` when you run `rails g controller`

File: features/steps/product_listing_steps.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
+Given("the following product exists") do |table|
+    table.hashes.each do |product|
+        Product.create!(product)
```
Comment (tochman) [inline features/steps/product_listing_steps.rb:3]: Please use Factories

**Pull request merged**